_posts/2019-01-02-Deviceid-Reliability.md

```diff
@@ -0,0 +1,95 @@
+
+It is expected that:
+- behind a deviceID there should be a real physical device used by a human person, 
+- when the a physical device uses different apps, those should give the same deviceID (unless it is resetted by the device's user which definetively does not occur at a per minute/hour frequency).
+
+Here we want to discover the deviceID which looks like real data but after some statistical studies you can categorize them as non reliable/useful in the sense that it is not guarantee that a real person is using the device.
+Dealing with null missing values or out-of-range values it is not what we are talking about.
+
+The data can be described as a collection of deviceIDs which visit an app at a certain time.
+<p align="center"> <img src="../images/Deviceid-Reliability/table_data_example.png"> </p>
+
+
+Let's go into details of the discovered patterns:
+
+* **Fraction of unique devices**<br/>
+Only apps with a certain number of distinct deviceIDs must be considered to avoid statistical fluctuations.
+With the histogram of the fraction of unique devices we can see both apps which are visited by too many/few distinct deviceIDs as also apps which shows some regular traffic.
+![frac_uniq_dev](../images/Deviceid-Reliability/fraction_unique_devices.png)
+The peaks in at one are those which are visited by always distinct deviceid while the other agglomerate close to zero.
+
+We suspect that something is broken in the way the apps provide/access the deviceID information for those apps.
+All deviceIDs which appear in a non reliable flagged app are considered non reliable, a simple inner join over the selected apps is enought to create the blacklist.
+
+
+* **Non realistic number of apps**<br/>
+For a fixed amount of time (e.g. 1 day) it is possible to observe the number of apps visited by each deviceID.
+From this plot we found out some clear outliers which cannot correspond to real human users.
+![num_uniqu_apps](../images/Deviceid-Reliability/num_unique_apps.png)
+In the plot the number of the vistited apps is standardized (the absolute value minus the overall mean divided by the standard deviation), so that the presence of outliers is evident.
+
+
+* **Short period of time**:<br/>
+When we observe a deviceID on our data for a very short period of time (say 1 minute) but we do not register any other occurrence of the deviceID over several days both in the past and in the future, we do not consider these deviceIDs to be worth of further analysis.
+In fact, it is not sure that we have a real person behind it, for example a machine which automatically reset the value would produce the same data.
+![one_over_more_days](../images/Deviceid-Reliability/only_1_observations_over_days.png)
+The plot above represents the observation of a deviceID in a 5 days period, both for users which shown a regular activity (d6, d7, d8) and for those which appear only once (d1, d2, d3).
```

Comment:
For me, it is not clear, what the y-axis depicts in the plot above. Explain a bit better, what I see here.